.github/workflows/auto_updates.yml

```diff
@@ -0,0 +1,71 @@
+# Workflow for updating Rust dependencies with Cargo and Javascript dependencies with npm.
+# Major version updates are handled separately, by Dependabot.
+---
+name: Update Deps
+
+on:
+  workflow_dispatch:
+  # Run every Monday at 0530 UTC
+  schedule:
+    - cron: '30 5 * * 1'
+
+jobs:
+  workflow-auto-updates:
+    name: Update dependencies
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        shell: bash
+    steps:
+      - name: Checkout the repo
+        uses: actions/checkout@c85c95e3d7251135ab7dc9ce3241c5835cc595a9 # v3.5.3
+
+      # This GPG key is for the `phylum-bot` account and used in order to ensure commits are signed/verified
+      - name: Import GPG key for bot account
+        uses: crazy-max/ghaction-import-gpg@72b6676b71ab476b77e676928516f6982eef7a41 # v5.3.0
```

Comment:
This action was reviewed and vetted the first time it was used in the `phylum-dev/phylum-ci` repository. It was found to be safe and useful for ensuring commits made in workflows are signed, without having to recreate the same features in script blocks and post-workflow jobs/steps.

One of it's nicest features is how it will automatically purge the imported GPG key and cache information and also kill the agent from the runner. It is possible to do this manually, but much more error prone. Here is an example of what it looks like for the single action to take care of both the setup and tear down:

<img width="887" alt="image" src="https://github.com/phylum-dev/cli/assets/18729796/26e167c2-02c8-4d3a-aa45-8d59b43e7c3c">

---

Further, the action is pinned to a specific hash and updates are only made in a Dependabot PR, providing a chance to ensure nothing sketchy is added to newer versions. "crazy-max" is a bit of an odd choice for someone asking folks to run code from a stranger...but this developer has _a lot_ of popular GitHub actions.

Reply:
> Further, the action is pinned to a specific hash and updates are only made in a Dependabot PR, providing a chance to ensure nothing sketchy is added to newer versions.

Do we actually have someone willing or capable of reviewing this action though? Did we actually review it to ensure no existing vulnerabilities are present?

I do not think I'd be capable of auditing updates for this without **significant** effort, so if someone else is able to I'm happy to leave these dependabot PRs up to them.

Reply:
The action was reviewed initially and then incrementally as it was updated via Dependabot PRs. It is true that this is a particularly sensitive action, but it is also true that any of the other actions or dependencies in use can similarly "go bad." Pinning actions/dependencies and reviewing their changes...through automation or manually...is the price to pay for vigilance.

Even so, further discussion with @louislang revealed that there is _not_ a requirement to have all intermediate commits signed. With the squash merge strategy that this repo has, most (all?) commits coming from PRs merged to `main` will show as signed by GitHub and it's verified signature (currently with a GPG key id of `4AEE18F83AFDEB23`) instead of the individual author(s) of the PR.

Therefore, the action has been removed in favor of making unsigned commits in workflows with the `phylum-bot` account.